### PR TITLE
Feature/robot  spring20fix

### PR DESCRIPTION
**Pull request by skristem**
# Critical Changes

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: robot/Cumulus/tests/browser/relationships/create_relationships.robot [hunk 1]
```diff
@@ -48,7 +48,8 @@ Create Relationships for contacts
     ...                                 &{contact2}[FirstName] &{contact2}[LastName]
     ...                                 Parent
 
-    Click Link                          link=Show more actions
+    Click More Actions Button
+    # Click Link                          link=Show more actions
```
Comment (smparedes) [inline robot/Cumulus/tests/browser/relationships/create_relationships.robot]: As mentioned before another comment, that could be removed.

File: robot/Cumulus/tests/browser/relationships/create_relationships.robot [hunk 2]
```diff
@@ -63,7 +64,6 @@ Create Relationships for contacts
     # Load Related List                   Relationships
```
Comment (smparedes) [inline robot/Cumulus/tests/browser/relationships/create_relationships.robot]: As mentioned earlier this is a comment that could be removed.

File: robot/Cumulus/tests/browser/tdtm/tdtm_username_exclude.robot.robot [hunk 3]
```diff
@@ -34,21 +38,25 @@ Update a Trigger Handler to Exclude a Username
     @{scratchuser} =             Salesforce Query               User
     ...                          select=Username
     ...                          Name=User User
-    Go To Record Home            ${triggerhandler}[0][Id]
+    Go To Page                   Details            Trigger_Handler__c            object_id=${triggerhandler}[0][Id]
+    # Go To Record Home            ${triggerhandler}[0][Id]
```
Comment (smparedes) [inline robot/Cumulus/tests/browser/tdtm/tdtm_username_exclude.robot.robot]: As mentioned before, another comment that could be removed.

File: robot/Cumulus/tests/browser/tdtm/tdtm_username_exclude.robot.robot [hunk 4]
```diff
@@ -34,21 +38,25 @@ Update a Trigger Handler to Exclude a Username
...
     ${uppercaseusername} =       Convert To Uppercase           ${scratchuser}[0][Username]
     API Modify Trigger Handler   ${triggerhandler}[0][Id]       ${ns}Usernames_to_Exclude__c=${uppercaseusername}
 
     # Create a Recurring Donation and verify no Opportunities are created
     &{contact} =                 API Create Contact             Email=jjoseph@robot.com
-    Go To Record Home            &{contact}[Id]
+    Go To Page                   Details            Contact           object_id=&{contact}[Id]
+    # Go To Record Home            &{contact}[Id]
```
Comment (smparedes) [inline robot/Cumulus/tests/browser/tdtm/tdtm_username_exclude.robot.robot]: As mentioned before, another comment that could be removed.

File: robot/Cumulus/tests/browser/tdtm/tdtm_username_exclude.robot.robot [hunk 5]
```diff
@@ -34,21 +38,25 @@ Update a Trigger Handler to Exclude a Username
...
     &{recurringdonation} =       API Create Recurring Donation  npe03__Contact__c=&{contact}[Id]
     ...                          Name=Julian Recurring Donation
     ...                          npe03__Amount__c=1200
     ...                          npe03__Installments__c=12
     ...                          npe03__Schedule_Type__c=Divide By
     ...                          npe03__Installment_Period__c=Monthly
-    Go To Record Home            &{recurringdonation}[Id]
-    Go To Record Home            &{contact}[Id]
+    Go To Page                   Details           npe03__Recurring_Donation__c           object_id=&{recurringdonation}[Id]
+    # Go To Record Home            &{recurringdonation}[Id]
```
Comment (smparedes) [inline robot/Cumulus/tests/browser/tdtm/tdtm_username_exclude.robot.robot]: As mentioned before another comment, that could be removed.

File: robot/Cumulus/tests/browser/tdtm/tdtm_username_exclude.robot.robot [hunk 6]
```diff
@@ -34,21 +38,25 @@ Update a Trigger Handler to Exclude a Username
...
+    Go To Page                   Details           Contact           object_id=&{contact}[Id]
+    # Go To Record Home            &{contact}[Id]
```
Comment (smparedes) [inline robot/Cumulus/tests/browser/tdtm/tdtm_username_exclude.robot.robot]: As mentioned before, another comment that could be removed.

File: robot/Cumulus/resources/AccountPageObject.py [hunk 7]
```diff
@@ -18,7 +18,6 @@ def _is_current_page(self):
         """ Verify we are on the Account detail page
             by verifying that the url contains '/view'
         """
-        self.selenium.wait_until_location_contains("/view", timeout=60, message="Detail view did not open in 1 min")
-        self.selenium.location_should_contain("/lightning/r/Account/",message="Current page is not an Account record detail view")
+        self.selenium.wait_until_location_contains("/lightning/r/Account/",message="Current page is not an Account record detail view")
```
[the remark this answers is not in the document]
Comment (skristem) [reply robot/Cumulus/resources/AccountPageObject.py]: ya I missed on this one, thanks will update

File: robot/Cumulus/resources/DataImportPageObject.py [hunk 8]
```diff
@@ -34,20 +35,30 @@ def open_data_import_record(self,di_name):
...
     
+    def _is_current_page(self):
+        """
+        Waits for the current page to be a Data Import list view
+        """
+        self.selenium.wait_until_location_contains("/view",timeout=60, message="Record detail view did not load in 1 min")
+        self.selenium.location_should_contain("DataImport__c",message="Current page is not a DataImport detail view")
     
-        
+    @capture_screenshot_on_error
```
[the remark this answers is not in the document]
Comment (skristem) [reply robot/Cumulus/resources/DataImportPageObject.py]: I have added that decorator on keywords that I have noticed issues with or hard to troubleshoot without a screenshot.

File: robot/Cumulus/resources/EngagementPlanTemplatesPageObject.py [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+class EngagementPlanDetailPage(BaseNPSPPage, DetailPage):
+
+    
+    def _is_current_page(self):
+        """
+        Waits for the current page to be a Data Import list view
+        """
+        self.selenium.wait_until_location_contains("/view",timeout=60, message="Records detail view did not load in 1 min")
+        self.selenium.location_should_contain("Engagement_Plan_Template__c",message="Current page is not an Engagement Plan record view")
+
```
[the remark this answers is not in the document]
Comment (skristem) [reply robot/Cumulus/resources/EngagementPlanTemplatesPageObject.py]: yes for now. We are refactoring Eng plan test cases on a seperate PR and there will be more code on that PR. As for this PR, this is just to fix spring20 issues so added only what is needed for now.

File: robot/Cumulus/resources/EngagementPlanTemplatesPageObject.py [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+class EngagementPlanListPage(BaseNPSPPage, ListingPage):
+
+    
+    def _is_current_page(self):
+        """
+        Waits for the current page to be a Data Import list view
+        """
+        self.selenium.wait_until_location_contains("/list",timeout=60, message="Records list view did not load in 1 min")
+        self.selenium.location_should_contain("Engagement_Plan_Template__c",message="Current page is not an Engagement Plan List view")
+
```
[the remark this answers is not in the document]
Comment (skristem) [reply robot/Cumulus/resources/EngagementPlanTemplatesPageObject.py]: just left some space to add more keywords later, not for any particular reason

File: robot/Cumulus/resources/GAUPageObject.py [hunk 11]
```diff
@@ -0,0 +1,24 @@
+from cumulusci.robotframework.pageobjects import ListingPage
+from cumulusci.robotframework.pageobjects import DetailPage
```
[the remark this answers is not in the document]
Comment (skristem) [reply robot/Cumulus/resources/GAUPageObject.py]: yup added first and removed but forgot this import, will remove it

File: robot/Cumulus/resources/AffiliationPageObject.py [hunk 12]
```diff
@@ -21,4 +21,9 @@ def _is_current_page(self):
         self.selenium.wait_until_location_contains("/view", timeout=60, message="Record view did not open in 1 min")
         self.selenium.location_should_contain("/lightning/r/npe5__Affiliation__c/",message="Current page is not an Affiliation record view")
```
[the remark this answers is not in the document]
Comment (skristem) [reply robot/Cumulus/resources/AffiliationPageObject.py]: didn't think of that, thanks for the suggestion

File: robot/Cumulus/resources/ManageHouseHoldPageObject.py [hunk 13]
```diff
@@ -34,6 +35,7 @@ def change_address_using(self, option, **kwargs):
            self.selenium.click_button("Save")
         self.selenium.unselect_frame()
```
[the remark this answers is not in the document]
Comment (skristem) [reply robot/Cumulus/resources/ManageHouseHoldPageObject.py]: that was added by ratna in a different PR not sure if there is a reason for having it outside an if statement. @rjanjanam was that intentional or misaligned?

File: robot/Cumulus/resources/NPSP.robot [hunk 14]
```diff
@@ -210,68 +211,72 @@ Create Level
...
-    Open NPSP Settings  Bulk Data Processes  Rollup Donations Batch
-    Click Settings Button    idPanelOppBatch    Run Batch
+    Open NPSP Settings          Bulk Data Processes                Rollup Donations Batch
+    Click Settings Button       idPanelOppBatch                    Run Batch
     # Wait For Locator    npsp_settings.status    CRLP_Account_SoftCredit_BATCH    Completed
     # Wait For Locator    npsp_settings.status    CRLP_RD_BATCH    Completed
     # Wait For Locator    npsp_settings.status    CRLP_Account_AccSoftCredit_BATCH    Completed
     # Wait For Locator    npsp_settings.status    CRLP_Contact_SoftCredit_BATCH    Completed
     # Wait For Locator    npsp_settings.status    CRLP_Account_BATCH    Completed
     # Wait For Locator    npsp_settings.status    CRLP_Contact_BATCH    Completed
```
[the remark this answers is not in the document]
Comment (skristem) [reply robot/Cumulus/resources/NPSP.robot]: batch names change if CRLPS is enabled. If a QA want to run tests on local with these enabled then they can switch. others I just forgot to delete after testing. Removing them now

File: robot/Cumulus/resources/NPSP.robot [hunk 15]
```diff
@@ -210,68 +211,72 @@ Create Level
...
     Select Frame And Click Element    ${frame_name}    ${ele_path}     @{others}
     
-Click Field And Select Date
-    [Arguments]    ${field}    ${date}
-    Click Element With Locator    bge.field-input    ${field}    
-    Click BGE Button    ${date}    
+# Click Field And Select Date
+    # [Arguments]    ${field}    ${date}
+    # Click Element With Locator    bge.field-input    ${field}    
+    # Click BGE Button    ${date}
```
[the remark this answers is not in the document]
Comment (skristem) [reply robot/Cumulus/resources/NPSP.robot]: forgot to delete. I always comment code and test and then delete once I confirm everything is working. Missed a few as there is lot of stuff in this PR

File: robot/Cumulus/tests/browser/donations_payments/data_imports.robot [hunk 16]
```diff
@@ -24,15 +26,16 @@ Data Imports
...
+    # Select Object Dropdown
+    # Click Link    link=All
+    # #Sleep    2
+    # #Click Link    link=Status
+    # Click Special Object Button       Start Data Import
+    # Wait For Locator    frame    NPSP Data Import
+    # # Select Frame With Title   NPSP Data Import
+    # # Click Button With Value   Begin Data Import Process
+    # Click Data Import Button    NPSP Data Import    button    Begin Data Import Process
+    # Wait For Batch To Process    BDI_DataImport_BATCH    Completed
```
[the remark this answers is not in the document]
Comment (skristem) [reply robot/Cumulus/tests/browser/donations_payments/data_imports.robot]: deleted

File: robot/Cumulus/tests/browser/relationships/create_relationships.robot (hunk 1 above)
Comment (skristem) [reply robot/Cumulus/tests/browser/relationships/create_relationships.robot]: done

File: robot/Cumulus/tests/browser/relationships/create_relationships.robot (hunk 2 above)
Comment (skristem) [reply robot/Cumulus/tests/browser/relationships/create_relationships.robot]: done

File: robot/Cumulus/resources/NPSP.py [hunk 17]
```diff
@@ -462,6 +497,17 @@ def validate_related_record_count(self,title,value):
             exp_value, actual_value
         )  
         
+    def verify_occurence(self,title,value):
```
Comment (rjanjanam) [inline robot/Cumulus/resources/NPSP.py]: Thought we changed this keyword to Validate Related Record Count . Is this keyword still needed ?

File: robot/Cumulus/resources/NPSP.py [hunk 18]
```diff
@@ -1220,11 +1289,11 @@ def verify_record_is_created_in_database(self,object_name,id):
         self.builtin.should_not_be_empty(record, msg="The database object {} with id {} is not in the database".format(object_name,id))
         return record
     
-
+    @capture_screenshot_on_error
```
Comment (rjanjanam) [inline robot/Cumulus/resources/NPSP.py]: Good usage for debugging .

**Pull request merged**